src/test/java/juja/microservices/keepers/controller/KeepersControllerTest.java

```diff
@@ -41,13 +46,48 @@
     private KeepersService service;
 
     @Inject
-    @InjectMocks
     private KeepersController controller;
 
     @Test
+    public void inactiveKeeperBadRequestTest() throws Exception {
+        //given
+        String badJsonRequest = "{\"from\":\"admin\",\"direction\":\"LMS\"}";
+
+        //then
+        mockMvc.perform(put("/keepers")
+                .contentType(APPLICATION_JSON_UTF8)
+                .content(badJsonRequest))
+                .andExpect(content().contentType(APPLICATION_JSON_UTF8))
+                .andExpect(status().isBadRequest());
+
+    }
+
+    @Test
+    public void inactiveKeeperSuccessTest() throws Exception {
+        KeeperRequest keeperRequest = new KeeperRequest("from", "uuid", "direction");
+        List<String> ids = new ArrayList<>();
+        ids.add("uuid");
+        when(service.inactiveKeeper(any(KeeperRequest.class))).thenReturn(ids);
+
+        String actual = mockMvc.perform(put("/keepers")
+                .contentType(APPLICATION_JSON_UTF8)
+                .content(toJson(keeperRequest)))
+                .andExpect(content().contentType(APPLICATION_JSON_UTF8))
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        verify(service).inactiveKeeper(argThat(reflectionEqual(keeperRequest)));
```

Comment:
in this case i just verify that correct keeperRequest came, so i wrote custom matcher wich  asserts two objects/collection(reflection is used to compare all fields of these values)  are equal. we could use it for all obj in future.